### PR TITLE
[docs] fix CDN usage & bundle externals

### LINKED ISSUE #3262: Blueprint Starter Kit: Cannot read property '__extends' of undefined
#### Environment

- __Package version(s)__: 
```
    classnames@^2.2
    dom4@^1.8
    react@^16.2.0/umd/react.production.min.js
    react-dom@^16.2.0/umd/react-dom.production.min.js
    react-transition-group@^2.2.1/dist/react-transition-group.min.js
    popper.js@^1.14.1/dist/umd/popper.js
    react-popper@^1.0.0/dist/index.umd.min.js
    @blueprintjs/core@^3.0.0
    @blueprintjs/icons@^3.0.0
```
- __Browser and OS versions__: 
Chrome Browser: Version 71.0.3578.98 (Official Build) (64-bit)
Mac OS 10.13.1

#### Steps to reproduce

1. Head to blueprint getting started docs: 
https://blueprintjs.com/docs/#blueprint/getting-started
2. Copy paste the example under CDN consumption into an index.html file
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width">
    <title>Blueprint Starter Kit</title>
    <link href="https://unpkg.com/normalize.css@^7.0.0" rel="stylesheet" />
    <link href="https://unpkg.com/@blueprintjs/core@^3.0.0/lib/css/blueprint.css" rel="stylesheet" />
    <link href="https://unpkg.com/@blueprintjs/icons@^3.0.0/lib/css/blueprint-icons.css" rel="stylesheet" />
  </head>
  <body>
    <script src="https://unpkg.com/classnames@^2.2"></script> 
    <script src="https://unpkg.com/dom4@^1.8"></script> 
    <script src="https://unpkg.com/react@^16.2.0/umd/react.production.min.js"></script> 
    <script src="https://unpkg.com/react-dom@^16.2.0/umd/react-dom.production.min.js"></script> 
    <script src="https://unpkg.com/react-transition-group@^2.2.1/dist/react-transition-group.min.js"></script> 
    <script src="https://unpkg.com/popper.js@^1.14.1/dist/umd/popper.js"></script> 
    <script src="https://unpkg.com/react-popper@^1.0.0/dist/index.umd.min.js"></script> 
    <script src="https://unpkg.com/@blueprintjs/core@^3.0.0"></script> 
    <script src="https://unpkg.com/@blueprintjs/icons@^3.0.0"></script> 
 
    <div id="btn"></div>
    <script>
      const button = React.createElement(Blueprint.Core.Button, {
        icon: "cloud",
        text: "CDN Blueprint is go!",
      });
      ReactDOM.render(button, document.querySelector("#btn"));
    </script> 
  </body>
</html>
```
3. Open index.html file in chrome or run in a local server like `serve` 
4. Observe the following exception in the console:
```
core.bundle.js:1 Uncaught TypeError: Cannot read property '__extends' of undefined
    at core.bundle.js:1
    at Object.<anonymous> (core.bundle.js:1)
    at n (core.bundle.js:1)
    at Object.<anonymous> (core.bundle.js:1)
    at n (core.bundle.js:1)
    at core.bundle.js:1
    at core.bundle.js:1
    at core.bundle.js:1
    at core.bundle.js:1
```

#### Actual behavior

The example Blueprint Quick Start for CDN consumption does not work. An exception is thrown in the console and no button renders on the screen. 

#### Expected behavior

Running the Example Blueprint Quick Start in the browser should render a button and not throw exceptions.

#### Possible solution

I'm not sure. Seems like perhaps some dependency is missing 🤷‍♂️

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- update webpack externals list to latest dependencies
- update CDN usage example to list correct required deps.
- push CDN usage section to bottom of Getting Started page to keep the focus on most common usage scenario (NPM)
- small edits to getting started language

### CODE REVIEW

File: packages/webpack-build-scripts/externals.js [hunk 1]
```diff
@@ -16,11 +16,14 @@ module.exports = externalize({
...
+    "popper.js": "Popper",
     "react": "React",
-    "react-transition-group": "ReactTransitionGroup",
     "react-day-picker": "DayPicker",
     "react-dom": "ReactDOM",
-    "tslib": "tslib",
+    "react-popper": "ReactPopper",
+    "react-transition-group": "ReactTransitionGroup",
+    "resize-observer-polyfill": "ResizeObserver",
+    "tslib": "window",
```
Comment (giladgray) [inline packages/webpack-build-scripts/externals.js]: i think _this_ is the line that actually fixes #3262

Comment (adidahiya) [reply packages/webpack-build-scripts/externals.js]: how does this work? I see very little documentation about this "window" option

File: packages/webpack-build-scripts/externals.js [hunk 2]
```diff
@@ -16,11 +16,14 @@ module.exports = externalize({
     "jquery": "$",
```
Comment (adidahiya) [inline packages/webpack-build-scripts/externals.js]: remove this, while you're at it?

File: packages/webpack-build-scripts/externals.js [hunk 3]
```diff
@@ -16,11 +16,14 @@ module.exports = externalize({
...
     "moment-timezone": "moment",
+    "popper.js": "Popper",
     "react": "React",
-    "react-transition-group": "ReactTransitionGroup",
     "react-day-picker": "DayPicker",
     "react-dom": "ReactDOM",
-    "tslib": "tslib",
+    "react-popper": "ReactPopper",
+    "react-transition-group": "ReactTransitionGroup",
+    "resize-observer-polyfill": "ResizeObserver",
```
Comment (adidahiya) [inline packages/webpack-build-scripts/externals.js]: this feels a little silly to require as an external... but I guess it's consistent with dom4

File: packages/webpack-build-scripts/externals.js (hunk 1 above)
Comment (giladgray) [reply packages/webpack-build-scripts/externals.js]: i had to dig through the tslib code to figure out what global it exposes. turns out it adds methods directly to `window`. same as dom4.

File: packages/webpack-build-scripts/externals.js (hunk 3 above)
Comment (giladgray) [reply packages/webpack-build-scripts/externals.js]: yah i could easily be convinced to remove this line and bundle this one lib, but might as well be consistent

**Pull request merged**